synop2bufr/__init__.py

```diff
@@ -1196,6 +1237,24 @@ def transform(data: str, metadata: str, year: int,
             LOGGER.error(f"Error parsing SYNOP report: {message}. {str(e)}")
             continue
 
+        # Now determine and load the appropriate mappings
+        # file depending on the value of the wind indicator.
+        # This will be updated for each message.
+        if msg['wind_indicator'] in [1, 4]:
+            # Use the new template if wind is measured
+            bufr_template = 307096
+            # Get mapping template, this needs to be
+            # reloaded everytime as each SYNOP can have a
+            # different number of replications
+            mapping = deepcopy(_mapping_307096)
+        else:
+            # Use the old template otherwise
```

Comment:
We need to map the wind indicator to the bitflag

```   
# Note bit 3 should never be set for synop, units of km/h not reportable
        if iw == 0:
            iw_translated = 0b0000  # Wind in m/s, default, no bits set
        elif iw == 1:
            iw_translated = 0b1000  # Wind in m/s with anemometer bit 1 (left most) set  # noqa
        elif iw == 3:
            iw_translated = 0b0100  # Wind in knots, bit 2 set
        elif iw == 4:
            iw_translated = 0b1100  # Wind in knots with anemometer, bits
            # 1 and 2 set # noq
        else:
            iw_translated = None  # 0b1111  # Missing value
    else:
        iw_translated = None  # 0b1111  # Missing value
```

Reply:
Doh! I'm not sure how I forgot this